nodl_python/nodl/index.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the
+# GNU Limited General Public License version 3, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranties of MERCHANTABILITY, SATISFACTORY QUALITY, or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Limited General Public License for more details.
+#
+# You should have received a copy of the GNU Limited General Public License along with
+# this program. If not, see <http://www.gnu.org/licenses/>.
+
+
+from pathlib import Path
+from typing import List, TYPE_CHECKING
+
+from ament_index_python.packages import get_package_share_directory
+
+from ._parsing import parse_multiple
+from .errors import NoNoDLFilesError
+
+if TYPE_CHECKING:
+    from .types import Node  # noqa: F401
+
+
+def get_nodl_xml_files_in_path(*, path: Path) -> List[Path]:
```

Comment:
This module is part of the public API right now, which means all these module-public functions are as well. Do we have solid reasons for that?

Reply:
These are consumed as-is from ros2nodl, which is technically an external package that includes nodl_python

Reply:
I have no idea what ros2nodl is, but I assume it's in the CLI PR? I think that can safely call private API; we can change it ourselves if we break it.

Reply:
Removed from public api in f2b0031